uast/transformer/positioner/positions.go

```diff
@@ -107,12 +114,23 @@ func fromUnicodeOffset(idx *positionIndex, pos *uast.Position) error {
 	return fromOffset(idx, pos)
 }
 
+func fromUTF16Offset(idx *positionIndex, pos *uast.Position) error {
+	off, err := idx.UTF16Offset(int(pos.Offset))
+	if err != nil {
+		return err
+	}
+	pos.Offset = uint32(off)
+	return fromOffset(idx, pos)
+}
+
 // runeSpan represents a sequence of UTF8 characters of the same size in bytes.
 type runeSpan struct {
-	firstRuneInd int // index of the first rune
-	byteOff      int // bytes offset of the first rune
-	numRunes     int // number of runes
-	runeSize     int // in bytes
+	firstRuneInd  int // index of the first rune
+	firstUTF16Ind int // index of the first UTF-16 code point
```

Comment:
Maybe I have missed something, but I find having these two separate is confusing: If there are no UTF-16 code points, how will this stay in sync? Or to put it another way, if someone requests the UTF-16 offset of a rune in a string with no UTF-16 code points, the handling of conversion _to_ surrogate form would need to be addressed. 

I think at least we should make it clear what the invariants are.

Reply:
Mentioned both offset and size invariants. Hope it helps.